### PR TITLE
Hide "Browse Media" button for unavailable media players

### LINKED ISSUE #20531: Media Browser button not disabled on unavailable media players in more info dialog
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

More info dialog for media players that are in unavailable state should disable the media browser button since it'll always fail.

Currently it lets you click it and _then_ shows you an error dialog, repeating information already shown on the more info dialog itself (that media player is unavailable). Feels like odd behavior when the user can just be prevented from clicking a non-functional button in the first place.

Screen Shot 2024-04-15 at 10 30 00 PM

Screen Shot 2024-04-15 at 10 30 08 PM

### Describe the behavior you expected

Media browser button should be disabled when media player entity is in unavailable state

### Steps to reproduce the issue

1. Have an unavailable entity (easy to reproduce with Plex clients)
2. open the more info of that unavailable media player, and then click "Browse Media" button (which will of course not work)
3. Get an error dialog 

### What version of Home Assistant Core has the issue?

2024.4.3

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Chrome

### Which operating system are you using to run this browser?

macOS Sonoma

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change
Hide the button if the entity is not available. The same is already done for the UI elements such as source selection, volume buttons, ... if the media player is not available.

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #20531
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/dialogs/more-info/controls/more-info-media_player.ts [hunk 1]
```diff
@@ -62,7 +62,8 @@ class MoreInfoMediaPlayer extends LitElement {
                 `
               )}
         </div>
-        ${supportsFeature(stateObj, MediaPlayerEntityFeature.BROWSE_MEDIA)
+        ${stateActive(stateObj) &&
```
Comment (karwosts) [inline src/dialogs/more-info/controls/more-info-media_player.ts:65]: Is `stateActive` really the correct filter? Does that mean you can't browse in `standby`?

`!isUnavailableState` instead maybe?

Comment (spacegaier) [reply src/dialogs/more-info/controls/more-info-media_player.ts:65]: I initially had used `isUnavailableState`. If browsing in standby is a valid supported use case, then reverting back to `isUnavailableState` sounds reasonable.

Comment (bramkragten) [reply src/dialogs/more-info/controls/more-info-media_player.ts:65]: yeah I agree `isUnavailableState` is the correct check

File: src/dialogs/more-info/controls/more-info-media_player.ts [hunk 2]
```diff
@@ -62,7 +63,8 @@ class MoreInfoMediaPlayer extends LitElement {
                 `
               )}
         </div>
-        ${supportsFeature(stateObj, MediaPlayerEntityFeature.BROWSE_MEDIA)
+        ${!isUnavailableState(stateObj) &&
```
Comment (silamon) [inline src/dialogs/more-info/controls/more-info-media_player.ts:66]:
```suggestion
        ${!isUnavailableState(stateObj.state) &&
```
Ensure to pass the state and not the stateObj.

Comment (spacegaier) [reply src/dialogs/more-info/controls/more-info-media_player.ts:66]: Duh...I shouldn't push changes that late at night 🙈

**Review by silamon — APPROVED** [review]

**Pull request merged**